### ISSUE TITLE
[BUG] - Calamares 3.2.61 freezing at Summary module after pressing Install button

### ISSUE THREAD

**Issue by D3vil0p3r** [calamares/calamares#2060]
**Describe the bug**
I'm using Arch Linux with `kernel arch2-1 #1 SMP PREEMPT_DYNAMIC x86_64 GNU/Linux` (installed by `linux` and `linux-headers` packages) and GNOME 42.4 with Wayland.

On my system I always used Calamares with custom modules in a correct manner with no issues but in the last days I'm getting a strange issue: when I make a new ISO of my system by archiso and in the Live Environment I run Calamares, when I reach the "Summary" section and I click on "Install" button, the Calamares window does not go to the Slideshows window but remains in the "Summary" section:
![image](https://user-images.githubusercontent.com/83867734/196057524-45fd5655-36e2-4fd1-a482-08875181dcce.png)

Despite Calamares remains in "Summary" section, the installation goes well, indeed, by running `sudo calamares -D8 | sudo tee calamares.log` we can see the correct logs of the installation (see attached file `calamares.log`). [calamares.log](https://github.com/calamares/calamares/files/9795682/calamares.log)

At the end of the installation, Calamares GUI switches immediately to the End Installation window (with some weird overlapped view):
![image](https://user-images.githubusercontent.com/83867734/196057592-97483c92-46f3-4d64-81b4-8d2e1580d76f.png)

and here, by reducing the Calamares window and reopening again, I can click on "Done" and the system is rebooted to access correctly to the installed system.

The main problem is that, since Calamares GUI, during the installation, stays on "Summary" section, a standard user could think that the system is hanging, but this is not true because I can move the Calamares window by the mouse and the logs are writing on the terminal.

My configuration worked for months, just in the last days, where I also performed a `sudo pacman -Syyu`, I started to have this issue.

I also rebuilt my Calamares configuration in a clean Arch Linux machine with the latest dependencies. The PKGBUILD of my Calamares configuration can be found [here](https://github.com/Athena-OS/athena-repository/blob/main/x86_64/packages/athena-calamares/PKGBUILD).

The custom modules used in `/etc/calamares` can be found [here](https://github.com/Athena-OS/athena-repository/tree/main/x86_64/packages/athena-calamares-config)

Probably the issue could be caused by some updated dependencies.

**To Reproduce**
1- Set up a new Arch Linux machine full upgraded.
2- Compile and create a package of Calamares by the configuration [here](https://github.com/Athena-OS/athena-repository/blob/main/x86_64/packages/athena-calamares/PKGBUILD) (here you can `makepkg -si` for installing the package directly on your system).
3- Add the custom modules on `/etc/calamares` and `/usr/lib/calamares/modules` that can be found [here](https://github.com/Athena-OS/athena-repository/tree/main/x86_64/packages/athena-calamares-config) (here you can `makepkg -si` for installing the package directly on your system)
4- Run `calamares_polkit` and set random choices until you reach the `Install` button in `Summary` section

**Expected behavior**
When the user clicks on `Install` button in `Summary` section, the Calamares GUI should move to the window with the installation progress bar instead to remain in the `Summary` section until the end of the installation.

**Comment by demmm** [comment]
There is way too much custom in these settings to debug this for you.  You will have to start with disabling all custom modules, see if summary then continues as should to the slideshow page, then add custom modules one by one until you see which one is the culprit (do so by removing those modules from the custom settings.conf).

**Comment by D3vil0p3r** [comment]
> There is way too much custom in these settings to debug this for you. You will have to start with disabling all custom modules, see if summary then continues as should to the slideshow page, then add custom modules one by one until you see which one is the culprit (do so by removing those modules from the custom settings.conf).

Thanks for the answer @demmm. For debugging, could be enough to just comment modules inside `/etc/calamares/settings.conf` or I should remove them physically from the disk?

Currently I'm also trying to remove libraries I used in my ISO for checking if some of them could cause this issue

**Comment by demmm** [comment]
No need to remove anything, just use the settings.conf (of course, remove from there), so the modules won't load.

**Comment by D3vil0p3r** [comment]
@demmm From `/etc/calamares/modules` I removed every custom module and I tested only with the following default Calamares modules:
* finished.conf
* fstab.conf
* initcpio.conf
* partition.conf
* removeuser.conf
* unpackfs.conf
* users.conf
* welcome.conf
The `settings.conf` file is:
```
modules-search: [ local ]

sequences:
- show:
  - welcome
  - locale
  - keyboard
  - partition
  - users
  - summary
- exec:
  - partition
  - mount
  - unpackfs
  - machineid
  - fstab
  - locale
  - keyboard
  - localecfg
  - luksbootkeyfile
  - luksopenswaphookcfg
  - initcpiocfg
  - initcpio
  - networkcfg
  - hwclock
  - services-systemd
  - removeuser
  - users
  - grubcfg
  - bootloader
  - umount
- show:
  - finished

branding: default
prompt-install: false
dont-chroot: false
oem-setup: false
disable-cancel: false
disable-cancel-during-exec: false
hide-back-and-next-during-exec: true
quit-at-end: false   
```
Here I tested also by `default` branding. Same issue: Calamares GUI freezes on Summary section and no "Install" section is shown on the GUI despite the installation is going well under the hood.

The ISO where this Calamares has been tested was created by Archiso with the following packages inside `packages.x86_64` file (I'm not sure if they can impact it or not):
```
amd-ucode
base
base-devel
btrfs-progs
dialog
dosfstools
edk2-shell
efibootmgr
grub
inetutils
intel-ucode
linux
linux-firmware
linux-headers
lvm2
man-db
man-pages
memtest86+
mesa
mesa-utils
mkinitcpio-archiso
mkinitcpio-nfs-utils
mtools
nano
nbd
netctl
networkmanager
os-prober
pv
rsync
squashfs-tools
sudo
syslinux
wireless_tools
wpa_supplicant

gdm
gnome

athena-calamares (it is the Calamares building I pasted in the first post with this PKGBUILD: https://github.com/Athena-OS/athena-repository/blob/main/x86_64/packages/athena-calamares/PKGBUILD)
athena-calamares-config (my custom configuration with this PKGBUILD: https://github.com/Athena-OS/athena-repository/tree/main/x86_64/packages/athena-calamares-config)
```
The ISO uses Chaotic AUR and BlackArch repositories. I don't think they are the issues because it worked for months.

I strongly think could be related to new updates of libraries like LibBoost and similar... I don't know why I'm getting that strange "graphical" behavior on the "Summary" section of Calamares GUI.

**Comment by D3vil0p3r** [comment]
@demmm do you have some `PKGBUILD` with Calamares 3.2.61 configuration that I can build on my machines and see if I get the same issue?

**Comment by demmm** [comment]
Boost updates will require rebuilds of anything that depends on it.  But this getting beyond the realm of a Calamares issue.  For your custom repos, make sure everything that links in any form to boost is rebuild.

Closing, since nothing points to this being a Calamares issue, and issues are not a help channel.
Please re-open if you can point to this being a Calamares bug.

**Comment by D3vil0p3r** [comment]
Also now tried Manjaro Calamares installer by [this PKGBUILD] (https://gitlab.manjaro.org/packages/extra/calamares/-/blob/master/PKGBUILD) on a clean Arch Linux machine and the result is this:
![image](https://user-images.githubusercontent.com/83867734/196273267-3dac7a86-8bad-44e9-b821-f5dfb2fcf545.png)

So I don't think this is related to my custom configuration. I reopen the ticket at least until I restrict the issue to only my configuration.

**Comment by D3vil0p3r** [comment]
I tested also Axyl Calamares on a pure Arch machine, also it same issue... so I can say it is not related to my configuration:

![image](https://user-images.githubusercontent.com/83867734/196292675-a29e0edf-fab3-46ff-804d-f61f272f290a.png)

**Comment by adriaandegroot** [comment]
You've got a QML slideshow configured, and in the log it's visible that the slideshow is advancing, but clearly it is not *showing* the slideshow. That could be a QML problem in your QML, or you've not got the Qt QML modules available, or .. many possibilities. As a first step, switch the slideshow (`branding.conf`) to show images instead of QML. That will tell you if the installation-progress-page component (the "slideshow", although it's a regular ViewStep) is working at all. Then dial back whatever branding you have to the default.

**Comment by dalto8** [comment]
Are you testing in a VM?  If so, do you see the same thing on bare metal?

**Comment by D3vil0p3r** [comment]
Hello @dalto8 as I said you yesterday in IRC chat, today I tested also EndeavourOS by referencing the [link](https://github.com/endeavouros-team/EndeavourOS-calamares) you provided me yesterday and by using the default EndeavourOS configuration and I get the same issue:
![image](https://user-images.githubusercontent.com/83867734/196504161-0d867ca2-b914-447d-a930-f92736bcd387.png)

I tested in a latest pure Arch Linux environment with GNOME 42 and Wayland (I tested also by Xorg) both in VM and bare metal.

@adriaandegroot According to my last tests, this Calamares issue is affecting GNOME, because I just tested it in the same environment I used for GNOME, but now I tested by XFCE and it is working well. So, I guess that some new updates related to some GNOME libraries could make that issue. I continue my investigation to detect which GNOME items have been updated and which of these could cause the issue.

Furthermore, I tested your suggestion about switching from `show.qml` to static image declaration on the `slideshow` field and it worked both on GNOME and XFCE. So, summarizing, the issue could be related Qt QML library. What is the Arch package managing Qt QML?
